app/controllers/api/v1/surveys_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class SurveysController < ApplicationController
+      include Pagy::Backend
+
+      def index
+        pagy, surveys = pagy_array(Survey.all, pagination_params)
+
+        render json: SurveySerializer.new(surveys, meta: meta_from_pagy(pagy))
+      rescue Pagy::OverflowError
+        render status: :not_found
+      end
+
+      def show
+        survey = Survey.find_by(id: params[:id])
+
+        if survey
+          render json: survey, include: %i[questions questions.answers]
+        else
+          render status: :not_found
+        end
+      end
+
+      private
+
+      def pagination_params
+        {
+          page: params.dig(:page, :number) || Pagy::VARS[:page],
+          items: params.dig(:page, :size)
+        }
+      end
+
+      def meta_from_pagy(pagy)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/3bdd1603c87059b5dda5aabfb14999fb437bbf29/app/controllers/api/v1/surveys_controller.rb#L35"></span>Doesn’t depend on instance state (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>